dataRetriever/txpool/interface.go

```diff
@@ -9,6 +9,8 @@ type txCache interface {
 	storage.Cacher
 
 	AddTx(tx *txcache.WrappedTransaction) (ok bool, added bool)
+	NotifyAccountNonce(accountKey []byte, nonce uint64)
+	ForgetAllAccountNonces()
```

Comment:
Ideally, we should have had a selective forgetfulness.

Reply:
does it cost too much to do that ?

Reply:
Implementation would be non-trivial. Forgetting all should not be seen as a performance issue, though (operation is fast) - indeed, selection in blocks following reverted ones might lead to a number of bad txs (depending on the content of the pool), but that corrects automatically in following blocks.

Anyway, the next optimization of the mempool removes the need of notifying / forgetting account nonces.